src/gdext/surface/properties.nim

```diff
@@ -119,131 +156,184 @@ template `@export_custom`*[T: SomeUserClass; S: SomeProperty](
       usage: set[PropertyUsageFlags] = PropertyUsageFlags.propertyUsageDefault;
     ): untyped =
   register_property(typedesc T, name, typedesc S, getter, setter,
-    hint, hintstring, usage)
+    hint, hintstring, usage) 
+
+macro gdexport_custom*(
+      iden: typed;
+      alias: static[alias] = noAlias;
+      hint: PropertyHint = propertyHintNone;
+      hintstring: String = gdstring"";
+      usage: set[PropertyUsageFlags] = PropertyUsageFlags.propertyUsageDefault;
+    ): untyped =
+    quote do:
+      register_property_iden(`iden`, `alias`, `hint`, `hintstring`, `usage`) 
 
-template `@export`*[T: SomeUserClass; S: SomeProperty](
+template gdexport*[T: SomeUserClass; S: SomeProperty](
     name;
     getter: proc(self: T): S;
     setter: proc(self: T; value: S)): untyped =
   register_property(typedesc T, name, typedesc S, getter, setter)
+macro gdexport*(iden: SomeProperty, alias: static[alias] = noAlias)  = 
+  let class = iden[0]
+  let propClass = iden.getTypeInst[1]
+  let propClassName = $propClass.toStrLit
+  let isEnum = propClass.getType.kind == nnkEnumTy
 
-template `@export_color_no_alpha`*[T: SomeUserClass; S: SomeProperty](
+  result = newStmtList()
+  if isEnum:
+    result.add registerEnumInternal(class, propClass, false)
+
+  result.add quote do: 
+    when `iden` is SomeObject:
+      register_property_iden(`iden`, `alias`, hint= propertyHintNodeType, hint_string= `propClassName`)
+    else:   
+      register_property_iden(`iden`, `alias`)
+    
+template gdexport_color_no_alpha*[T: SomeUserClass, S: SomeColor](
       name;
       getter: proc(self: T): S;
       setter: proc(self: T; value: S);
     ): untyped =
   register_property(typedesc T, name, typedesc S, getter, setter,
     hint= propertyHintColorNoAlpha)
+macro gdexport_color_no_alpha*(iden: SomeColor, alias: static[alias] = noAlias)  = quote do: 
+  register_property_iden(`iden`, `alias`, hint= propertyHintColorNoAlpha)
```

Comment:
I think it is better to change such macro that has only `quote do` block to template. I don't know if this will have a positive effect on performance, but it will be a bit easy to read.

Reply:
Agreed, and changed in the latest commit